src/main/java/com/commercetools/sync/inventory/InventorySync.java

```diff
@@ -0,0 +1,380 @@
+import io.sphere.sdk.inventory.InventoryEntryDraftBuilder;
+import io.sphere.sdk.models.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.commercetools.sync.inventory.utils.InventoryDraftTransformer.transformToDrafts;
+import static java.lang.String.format;
+
+/**
+ * Default implementation of inventories sync process.
+ */
+public final class InventorySync extends BaseSync<InventoryEntryDraft, InventoryEntry, InventorySyncStatistics,
+        InventorySyncOptions> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InventorySync.class);
+
+    /*
+     * Holds threads for executing sync process in parallel. Should be instantiated when parallel processing is
+     * enabled via InventorySyncOptions.
+     */
+    private ExecutorService executorService = null;
+
+    //Cache that maps supply channel key to supply channel Id for supply channels existing in database.
+    private Map<String, String> supplyChannelKeyToId;
+
+    private InventoryService inventoryService;
+
+    private TypeService typeService;
+
+    public InventorySync(@Nonnull final InventorySyncOptions syncOptions) {
+        this(syncOptions, new InventoryServiceImpl(syncOptions.getCtpClient().getClient()),
+                new TypeServiceImpl(syncOptions.getCtpClient().getClient()));
+    }
+
+    InventorySync(final InventorySyncOptions syncOptions, final InventoryService inventoryService,
+                  final TypeService typeService) {
+        super(new InventorySyncStatistics(), syncOptions, LOGGER);
+        this.inventoryService = inventoryService;
+        this.typeService = typeService;
+        if (syncOptions.getParallelProcessing() > 1) {
+            executorService = Executors.newFixedThreadPool(syncOptions.getParallelProcessing());
+        }
+    }
+
+    /**
+     * <p>Performs full process of synchronisation between inventory entries present in a system
+     * and passed {@code inventories}. This is accomplished by:
+     * <ul>
+     *     <li>Comparing entries and drafts by {@code sku} and {@code supplyChannel} key</li>
+     *     <li>Calculating of necessary updates and creation commands</li>
+     *     <li>Actually <strong>performing</strong> changes in a target CTP project by sphere calls</li>
+     * </ul>
+     * The process is customized according to {@link InventorySyncOptions} passed to constructor of this object.
+     * After the process finishes you can obtain its summary by calling {@link InventorySync#getStatistics()}.</p>
+     * <p><strong>This method is meant to be called once. For new sync process please create new {@link InventorySync}
```

Comment:
@heshamMassoud we should redesign this mutable statistic, like generate a new one for every sync process, but not reuse a dirty one